### ISSUE TITLE
Feature Reqeust: add a callback-based HTTP response health indicator

### ISSUE THREAD

**Issue by trescube** [nestjs/terminus#984]
(Okay, I admit that I failed to read the CONTRIBUTING.md guidelines first and already have a PR ready to go, which I'll submit shortly)

## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

For remote HTTP services that need to be health-checked beyond a simple ping, it would be great to have a HTTP response health indicator that passes the templatized Axios response to a callback that returns a `true` or `false` based on some condition of the response.  

## Expected behavior

See above

## Minimal reproduction of the problem with instructions
No problem

## What is the motivation / use case for changing the behavior?

For example, the `/_cat/health` endpoint for Elasticsearch returns an array of clusters with status of either `green`, `yellow`, or `red` where non-`green` should be considered "down" and none of the existing `HealthIndicator`s cover this .  I've written my own `HealthIndicator` instance to solve this but doing this for every instance of a nuanced HTTP service health is getting a bit tedious.  I realize that the elasticsearch library for js has a `.cat.health` function, but this example is for illustration purposes.   

Example Elasticsearch `/_cat/health` response:

```
curl -i "http://localhost:9200/_cat/health?format=json&pretty"
HTTP/1.1 200 OK
content-type: application/json; charset=UTF-8
content-length: 368
access-control-allow-credentials: true

[
  {
    "epoch" : "1609275011",
    "timestamp" : "20:50:11",
    "cluster" : "docker-cluster",
    "status" : "yellow",
    "node.total" : "1",
    "node.data" : "1",
    "shards" : "1",
    "pri" : "1",
    "relo" : "0",
    "init" : "0",
    "unassign" : "1",
    "pending_tasks" : "0",
    "max_task_wait_time" : "-",
    "active_shards_percent" : "50.0%"
  }
]
```

Note that the HTTP status is still 200 despite the cluster being not healthy (`green` is healthy, `yellow`/`red` are not).

## Environment

<pre><code>
Nest version: X.Y.Z

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:
</code></pre>

**Comment by BrunnerLivio** [comment]
There is a pre-release available for this [7.1.0-next.0](https://github.com/nestjs/terminus/releases/tag/7.1.0-next.0) :tada:

Install it with:

```
npm install @nestjs/terminus@7.1.0-next.0
```

**Comment by BrunnerLivio** [comment]
This has been now been released with [7.1.0](https://github.com/nestjs/terminus/releases/tag/7.1.0) :tada: :rocket:
